packages/Filters/src/Filters.js

```diff
@@ -0,0 +1,148 @@
+import React from "react";
+import PropTypes from "prop-types";
+import AddIcon from "@paprika/icon/lib/Add";
+import Button from "@paprika/button";
+import FilterIcon from "@paprika/icon/lib/Filter";
+import Panel from "@paprika/panel";
+import useI18n from "@paprika/l10n/lib/useI18n";
+import Item from "./components/Item";
+import FilterContext from "./context";
+import columnShape from "./columnShape";
+import * as types from "./types";
+
+import * as sc from "./Filters.styles";
+
+function getLabelText(numberOfFilters, I18n) {
+  switch (numberOfFilters) {
+    case 0:
+      return I18n.t("filters.label");
+    case 1:
+      return I18n.t("filters.singular_label");
+    default:
+      return I18n.t("filters.plural_label", { numberOfFilters });
+  }
+}
+
+export default function Filters(props) {
+  const {
+    appliedNumber,
+    children,
+    columns,
+    data,
+    onAddFilter,
+    onApply,
+    onCancel,
+    onChangeOperator,
+    onClear,
+    operator,
+    rulesByType,
+  } = props;
+  const I18n = useI18n();
+  const filtersRef = React.useRef(null);
+  const [isOpen, setIsOpen] = React.useState(false);
+
+  function handleClickTrigger() {
+    setIsOpen(prevIsOpen => !prevIsOpen);
+  }
+
+  function handleApply() {
+    onApply();
+    setIsOpen(false);
+  }
+
+  function handleCancel() {
+    setIsOpen(false);
+    onCancel();
+  }
+
+  function handleClear() {
+    onClear();
+  }
+
+  return (
+    <FilterContext.Provider value={{ filtersRef, columns, data, operator, onChangeOperator, rulesByType }}>
+      <Panel data-pka-anchor="filters.panel" isCompact isOpen={isOpen} onClose={handleCancel}>
+        <Panel.Header>
+          <sc.PanelHeaderWrapper>
+            <FilterIcon aria-hidden="true" />
+            {I18n.t("filters.label")}
+          </sc.PanelHeaderWrapper>
+        </Panel.Header>
+
+        <Panel.Trigger
+          data-pka-anchor="filters.panelTrigger"
+          icon={<FilterIcon />}
+          isSemantic={false}
+          kind={Button.types.kind.FLAT}
+          onClick={handleClickTrigger}
+        >
+          {getLabelText(appliedNumber, I18n)}
+        </Panel.Trigger>
+        <sc.FiltersPanel ref={filtersRef}>
+          {React.Children.count(children) === 0 ? (
+            <sc.GenericNoAppliedPlaceholder>{I18n.t("filters.no_filters_applied")}</sc.GenericNoAppliedPlaceholder>
+          ) : (
+            children
+          )}
+          <Button data-pka-anchor="filters.addFilterButton" icon={<AddIcon />} onClick={onAddFilter}>
+            {I18n.t("filters.actions.add")}
+          </Button>
+        </sc.FiltersPanel>
+        <Panel.Footer isSticky>
+          <sc.Footer>
+            <Button data-pka-anchor="filters.applyButton" onClick={handleApply} kind="primary">
```

Comment:
what are these `data-pka-achor`s for, a future test?

Reply:
Yes!